editoast/src/views/v2/scenario.rs

```diff
@@ -0,0 +1,620 @@
+
+impl From<ScenarioCreateForm> for Changeset<Scenario> {
+    fn from(scenario: ScenarioCreateForm) -> Self {
+        Scenario::changeset()
+            .name(scenario.name)
+            .description(scenario.description)
+            .creation_date(Utc::now().naive_utc())
+            .last_modification(Utc::now().naive_utc())
+            .infra_id(scenario.infra_id)
+            .timetable_id(scenario.timetable_id)
+            .tags(scenario.tags)
+    }
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "scenario")]
+#[allow(clippy::enum_variant_names)]
+pub enum ScenarioError {
+    /// Couldn't found the scenario with the given scenario ID
+
+    #[error("Scenario '{scenario_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { scenario_id: i64 },
+
+    #[error("Timetable '{timetable_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    TimetableNotFound { timetable_id: i64 },
+
+    #[error("Infra '{infra_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    InfraNotFound { infra_id: i64 },
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, ToSchema)]
+#[schema(as = ScenarioResponseV2)]
+pub struct ScenarioResponse {
+    #[serde(flatten)]
+    pub scenario: Scenario,
+    pub infra_name: String,
+    pub trains_count: i64,
+    pub project: Project,
+    pub study: Study,
+}
+
+impl ScenarioResponse {
+    pub fn new(
+        scenarios_with_details: ScenarioWithDetails,
+        project: Project,
+        study: Study,
+    ) -> Self {
+        Self {
+            scenario: scenarios_with_details.scenario,
+            infra_name: scenarios_with_details.infra_name,
+            trains_count: scenarios_with_details.trains_count,
+            project,
+            study,
+        }
+    }
+}
+
+/// Create a scenario
+#[utoipa::path(
+    tag = "scenariosv2",
+    params(ProjectIdParam, StudyIdParam),
+    request_body = ScenarioCreateFormV2,
+    responses(
+        (status = 201, body = ScenarioResponseV2, description = "The created scenario"),
+    )
+)]
+#[post("")]
+async fn create(
+    db_pool: Data<DbPool>,
+    data: Json<ScenarioCreateForm>,
+    path: Path<(i64, i64)>,
+) -> Result<Json<ScenarioResponse>> {
+    let (project_id, study_id) = path.into_inner();
+    let timetable_id = data.timetable_id;
+    let scenario: Changeset<Scenario> = data.into_inner().into();
+
+    let mut tx = db_pool.get().await?;
+
+    let scenarios_response = tx
+        .transaction::<_, InternalError, _>(|conn| {
+            async {
+                // Check if the project and the study exist
+                let (mut project, study) =
+                    check_project_study_conn(conn, project_id, study_id).await?;
+
+                // Check if the timetable exists
+                let _ = Timetable::retrieve_or_fail(conn, timetable_id, || {
+                    ScenarioError::TimetableNotFound { timetable_id }
+                })
+                .await?;
```

Comment:
We should do the same with the `infra_id`. Currently, we get a `500` if the infra doesn't exist.